shopinvader_anonymous_partner/models/res_partner.py

```diff
@@ -101,3 +102,31 @@ def _get_anonymous_partner__cookie(self, cookies: Cookies):
         if not token:
             return self.env["res.partner"].sudo().browse()
         return self._get_anonymous_partner__token(token)
+
+    def _promote_from_anonymous_partner(self, anonymous_partner: ResPartner):
+        """
+        Promote an anonymous partner to an authenticated partner.
+
+        This method should be overridden by other modules to implement
+        the partner resolution logic, merging the anonymous partner cart
+        for instance.
+
+        This method can return False to prevent the anonymous partner cookie
+        from being deleted.
+        """
+        return True
+
+    @api.model
+    def _promote_anonymous_partner(
+        self, partner: ResPartner, cookies: Cookies, response: Response
+    ):
+        """
+        Promote the current anonymous partner to the given authenticated partner.
+
+        This method calls the partner promotion and removes the anonymous partner cookie.
+        """
+        anonymous_partner = self._get_anonymous_partner__cookie(cookies)
+        if partner._promote_from_anonymous_partner(
+            anonymous_partner,
+        ):
```

Comment:
Not sure it's the right design.
Naturally I do not expect that the method "_promote_from_anonymous_partner" should return True or False to drop the cookie.
Do you have existing case where you do not want to drop the cookie?
If not => just always drop it (we will see what we need when we have the need)

(More over for the 0.01% of case that we want to change the behaviour we can still inherit the method _promote_anonymous_partner to pass something in the context and then use that key so skip the cookies deletation)

Reply:
I was thinking about a promotion failure where we don't want to drop the cookie but fair enough, I'll remove the return value.